pytorch_lightning/callbacks/lr_monitor.py

```diff
@@ -192,6 +209,23 @@ def _add_prefix(
         count = seen_optimizer_types[optimizer_cls]
         return name + f'-{count - 1}' if count > 1 else name
 
+    def _add_suffix(self, name: str, param_groups: List[Dict], param_group_index: int, use_names: bool = True) -> str:
+        if len(param_groups) > 1:
+            if not use_names:
+                return f'{name}/pg{param_group_index+1}'
+            else:
+                pg_name = param_groups[param_group_index].get('name', f'pg{param_group_index+1}')
+                return f'{name}/{pg_name}'
+        elif use_names:
+            pg_name = param_groups[param_group_index].get('name')
+            return f'{name}/{pg_name}' if pg_name else name
+        else:
+            return name
+
+    def _duplicate_param_group_names(self, param_groups: List[Dict]) -> Set[str]:
```

Comment:
Maybe use a `collections.Counter` for efficiency instead of generating the list and then calling count for each element

Reply:
See: https://github.com/PyTorchLightning/pytorch-lightning/pull/7987#discussion_r652874708 for my thoughts on this.

Reply:
You can see my thoughts on this here: https://github.com/PyTorchLightning/pytorch-lightning/pull/7987#discussion_r652874708

Reply:
So my proposal was the following

```python
    def _duplicate_param_group_names(self, param_groups: List[Dict]) -> Set[str]:
        names = [pg.get('name', f'pg{i}') for i, pg in enumerate(param_groups, start=1)]
        counter = Counter(names)
        return set() if len(counter) == len(names) else {k for k, v in counter.items() if v > 1}
```

But I guess the Counter performance does not really matter since the else will execute only before erroring out